trunk/SUAVE/Components/Energy/Converters/Propeller.py

```diff
@@ -112,41 +114,356 @@ def spin(self,conditions):
+
+
+        
+  
+        thrust[conditions.propulsion.throttle[:,0] <=0.0] = 0.0
+        power[conditions.propulsion.throttle[:,0]  <=0.0] = 0.0
+        
+        thrust[omega1<0.0] = - thrust[omega1<0.0]
+
+        etap     = V*thrust/power     
+        
+        conditions.propulsion.etap = etap
+        
+        # store data
+        results_conditions = Data     
+        outputs   = results_conditions(
+            n_blades                  = B,
+            R                         = R,
+            D                         = D,
+            number_sections           = N,
+            radius_distribution       = np.linspace(Rh ,R, N),
+            chord_distribution        = c,     
+            twist_distribution        = beta,            
+            r0                        = r,
+            thrust_angle              = theta,
+            speed_of_sound            = conditions.freestream.speed_of_sound,
+            density                   = conditions.freestream.density,
+            velocity                  = Vv, 
+            vt                        = vt, 
+            va                        = va, 
+            drag_coefficient          = Cd,
+            lift_coefficient          = Cl,       
+            omega                     = omega,          
+            
+            blade_dT_dR               = rho*(Gamma*(Wt-epsilon*Wa)),   
+            blade_dT_dr               = rho*(Gamma*(Wt-epsilon*Wa))*R,  
+            blade_T_distribution      = rho*(Gamma*(Wt-epsilon*Wa))*deltar, 
+            blade_T                   = thrust/B,  
+            Ct                        = 0, 
+            Cts                       = 0, 
+            
+            blade_dQ_dR               = rho*(Gamma*(Wa+epsilon*Wt)*r), 
+            blade_dQ_dr               = rho*(Gamma*(Wa+epsilon*Wt)*r)*R,
+            blade_Q_distribution      = rho*(Gamma*(Wa+epsilon*Wt)*r)*deltar,
+            blade_Q                   = torque/B,   
+            Cq                        = 0, 
+            
+            power                     = power,
+            
+            mid_chord_aligment        = self.mid_chord_aligment     
+        ) 
+        
+        return thrust, torque, power, Cp, outputs  , etap  
+
+
+    
+
+    def spin_variable_pitch(self,conditions):
+        """ Analyzes a propeller given geometry and operating conditions
+                 
+                 Inputs:
+                     hub radius
+                     tip radius
+                     rotation rate
+                     freestream velocity
+                     number of blades
+                     number of stations
+                     chord distribution
+                     twist distribution
+                     airfoil data
+       
+                 Outputs:
+                     Power coefficient
+                     Thrust coefficient
+                     
+                 Assumptions:
+                     Based on Qprop Theory document
```

Comment:
Can a more specific source be added?

Reply:
Proper header added

Reply:
Readded this comment in the right spot on the new files.